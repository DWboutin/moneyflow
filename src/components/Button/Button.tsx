import React, { ReactNode } from "react";
import styled from "styled-components";

import { ThemeContainer } from "../../styles/styles";
import { fontSizes } from "../../styles/typography";

const Container = styled.button`
  padding: 9px 40px 10px;
  color: ${({ theme }: ThemeContainer) => theme.button.normal.text};
  background-color: ${({ theme }: ThemeContainer) => theme.button.normal.bg};
  font-size: ${fontSizes.normal};
  border: 0px;
  border-radius: 20px;
  cursor: pointer;

  &:hover {
    background-color: ${({ theme }: ThemeContainer) => theme.button.hover.bg};
  }
  &:active {
    background-color: ${({ theme }: ThemeContainer) => theme.button.active.bg};
  }
`;

export type ButtonTypes = "submit" | "button";

interface Props {
  children: ReactNode;
  type?: ButtonTypes;
}

const Button: React.FunctionComponent<Props> = ({
  children,
  type = "button",
}) => {
  return (
    <Container type={type} role="button">
      {children}
    </Container>
  );
};

export default Button;
