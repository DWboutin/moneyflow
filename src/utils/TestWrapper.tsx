import React, { ReactNode } from "react";
import { Provider } from "react-redux";
import { ToastContainer } from "react-toastify";
import { ThemeProvider } from "styled-components";

import store from "../store/store";
import GlobalStyle from "../styles/globalStyle";

import { theme } from "../styles/styles";

interface Props {
  children: ReactNode;
}

const TestWrapper: React.FunctionComponent<Props> = ({ children }) => (
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
      <ToastContainer />
    </ThemeProvider>
  </Provider>
);

export default TestWrapper;
