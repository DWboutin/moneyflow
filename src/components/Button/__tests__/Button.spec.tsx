import React from "react";
import { render, RenderResult } from "@testing-library/react";

import Button, { Props } from "../Button";
import TestWrapper from "../../../utils/TestWrapper";

describe("<Button />", () => {
  let renderResult: RenderResult;

  const renderButton = ({ children, type }: Props) => {
    renderResult = render(
      <TestWrapper>
        <Button type={type}>{children}</Button>
      </TestWrapper>
    );
  };

  describe("default rendering", () => {
    beforeEach(() => {
      renderButton({ children: "Button label" });
    });

    it("should match snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });
  });

  describe("type='submit'", () => {
    beforeEach(() => {
      renderButton({ children: "Button label", type: "submit" });
    });

    it("should match snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });
  });
});
