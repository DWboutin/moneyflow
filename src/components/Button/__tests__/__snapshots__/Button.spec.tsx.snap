// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> default rendering should match snapshot 1`] = `
<div>
  <button
    class="sc-bdvvtL RYNZf"
    role="button"
    type="button"
  >
    Button label
  </button>
  <div
    class="Toastify"
  />
</div>
`;

exports[`<Button /> type='submit' should match snapshot 1`] = `
<div>
  <button
    class="sc-bdvvtL RYNZf"
    role="button"
    type="submit"
  >
    Button label
  </button>
  <div
    class="Toastify"
  />
</div>
`;
